src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,309 @@
+    private String dbHost = "localhost";
+    private int dbPort = 2222;
+    private int dbTimeout = 0;
+    private String dbPassword = "";
+
+    @Override
+    public void load() throws ClassNotFoundException, IOException {
+    }
+
+    @Override
+    public void unload() throws IOException {
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START Sessions
+    //------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public int getRejectedSessions() {
+        // Essentially do nothing.
+        return 0;
+    }
+
+    @Override
+    protected RedisSessionT8 getNewSession() {
+        return new RedisSessionT8(this);
+    }
+
+    @Override
+    public Session createSession(String requestedSessionId) {
+
+        RedisSessionT8 session = (RedisSessionT8) createEmptySession();
+
+        // Initialize the properties of the new session and return it
+        session.setId(UUID.randomUUID().toString());
+        session.setNew(true);
+        session.setValid(true);
+        session.setCreationTime(System.currentTimeMillis());
+        session.setMaxInactiveInterval(this.maxInactiveInterval);
+
+        try {
+            saveSession(session);
+        } catch (IOException e) {
+            log.error("Error - Context: createSession. Description: " + e.getMessage());
+        }
+
+        return session;
+    }
+
+    /**
+     * Add session to this manager
+     * @param session
+     */
+    @Override
+    public void add(Session session) {
+        try {
+            saveSession(session);
+        } catch (IOException e) {
+            log.error("Error - Context: add. Description: " + e.getMessage());
+        }
+    }
+
+
+    @Override
+    public Session findSession(String id) throws IOException {
+        try {
+
+            Hashtable<String, byte[]> result = withRedis(
+                    (Jedis jedis)-> {
+                        Hashtable<String, byte[]> r =  new Hashtable<>();
+                        r.put(REDIS_METADATA_KEY, jedis.get((id + REDIS_METADATA_KEY).getBytes()));
+                        r.put(REDIS_ATTRIBUTES_KEY, jedis.get((id + REDIS_ATTRIBUTES_KEY).getBytes()));
+                        return r;
+                    }
+            );
+
+            Hashtable<String, Object> metadata =
+                    (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(result.get(REDIS_METADATA_KEY)));
+
+            Hashtable<String, Object> attributes =
+                    (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(result.get(REDIS_ATTRIBUTES_KEY)));
+
+            return getSession(metadata, attributes);
+
+        } catch (Exception e) {
+            log.error("Error - Context: findSession. Description: " + e.getMessage());
+            throw new IOException("Session not found with id: " + id);
```

Comment:
This isn't an IO Exception, and you don't know that the session ID is the problem. This error message and error type are both bad.